File: fabric/src/main/java/me/lunaluna/fabric/elytrarecast/Timer.java
package me.lunaluna.fabric.elytrarecast;

import java.util.function.Supplier;

public class Timer {

    private Supplier<Integer> getCooldown;

    public Timer(Supplier<Integer> supplier) {
        this.getCooldown = supplier;
    }

    private long previousTime = 0;

    public void runOnCooldown(Runnable runnable) {
        var time = System.currentTimeMillis();
        var diff = time - previousTime;
        if (diff >= getCooldown.get()) {
            runnable.run();
            previousTime = time;
        }
    }
}

File: fabric/src/main/java/me/lunaluna/fabric/elytrarecast/Timer.kt
package me.lunaluna.fabric.elytrarecast

import java.util.function.Supplier

class Timer(val getCooldown: Supplier<Int>) {
    private var previousTime = 0L

    fun runOnCooldown(runnable: Runnable) {
        val time = System.currentTimeMillis()
        val diff = time - previousTime
        if (diff >= getCooldown.get()) {
            runnable.run()
            previousTime = time
        }
    }
}